forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/apiclient/ApiClientIdentityFactory.java

```diff
@@ -0,0 +1,224 @@
+                        String errorString = "Client presented an invalid Certificate " +
+                                "Type for use as a Transport certificate. Type presented ': " + certType + "'";
+                        log.debug("getApiClientIdentity() {}", errorString);
+                        throw new ApiClientException(errorString);
+
+                }
+            } else {
+                apiClientIdentity = createOBTransportIdentity(authentication);
+            }
+        } else if (principal instanceof X509Authentication){
+            X509Authentication authentication = (X509Authentication) principal;
+            apiClientIdentity = createOBTransportIdentity(authentication);
+        }
+        return apiClientIdentity;
+    }
+
+    private ApiClientIdentity createOBTransportIdentity(X509Authentication authentication)
+            throws OAuth2InvalidClientException, ApiClientException {
+        ApiClientCertificateType certType = getApiClientCertificateTypeFromX509(authentication);
+        if (certType == ApiClientCertificateType.OB_TRANSPORT) {
+            return new ApiClientIdentityOBTransport(authentication);
+        }
+        String errorString = "Client presented an invalid Certificate " +
+                "Type for use as a Transport certificate. Type presented ': " + certType + "'";
+        log.debug("getApiClientIdentity() {}", errorString);
+        throw new ApiClientException(errorString);
+    }
+
+    private ApiClientCertificateType getApiClientCertificateTypeFromX509(X509Authentication authentication)
+            throws ApiClientException {
+        ApiClientCertificateType type;
+        X509Certificate[] certChain = authentication.getCertificateChain();
+        String issuer = getTransportCertificateIssuer(certChain);
+        log.debug("getApiClientCertificateType() certificate issuer is '{}'", issuer);
+        if(issuer.equalsIgnoreCase(OBIE_ISSUER_NAME)){
+            List<String> certUsage = getTransportCertificateUsage(certChain);
+            if(certUsage.contains(CertificateUsageOIDs.SIGNER_OF_DOCUMENTS)){
+                type = ApiClientCertificateType.OB_SIGNING;
+            } else if (certUsage.contains(CertificateUsageOIDs.CLIENT_AUTH)){
+                type = ApiClientCertificateType.OB_TRANSPORT;
+            } else {
+                log.debug("getApiClientCertificateTypeFromX509() did not find expected extended key usage.");
+                throw new ApiClientException("OBIE issued (legacy) X509Certificate did not contain expected extended " +
+                        "certificate usage purposes. Contained " + certUsage + ". Expected " +
+                        CertificateUsageOIDs.CLIENT_AUTH + " or " + CertificateUsageOIDs.SIGNER_OF_DOCUMENTS);
+            }
+        } else {
+            String errorMessage = "Non eIDAS/PSD2 certificate presented, but it is not an OB issued certificate.";
+            log.debug("getApiClientCertificateTypeFromX509() {}", errorMessage);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.